doc/sphinx/language/extensions/implicit-arguments.rst

```diff
@@ -284,61 +288,129 @@ Declaring Implicit Arguments
+       evaulate to constructors.  See :tacn:`simpl`.
+
+     :n:`@name {? % @scope }`
+       a *formal parameter* of the function :n:`smart_qualid` (i.e.
+       the parameter name used in the function definition).  Unless `rename` is specified,
+       the list of :n:`@name`\s must be a prefix of the formal parameters, including all implicit
+       arguments.  `_` can be used to skip over a formal parameter.
+       :token:`scope` can be either a scope name or its delimiting key.
+
+     `clear implicits`
+       makes all implicit arguments into explicit arguments
+     `default implicits`
+       automatically determine the implicit arguments of the object.
+       See :ref:`auto_decl_implicit_args`.
+     `rename`
+       rename implicit arguments for the object
+     `assert`
+      assert that the object has the expected number of arguments with the
+      expected names.  See the example here: :ref:`renaming_implicit_arguments`.
+
+      .. todo: "Arguments Nat.add {n}%bool x%nat : assert." also changes n to {n}%nat.
+         feature or bug?  describe?
+
+     `clear scopes`
+       clears argument scopes of :n:`@smart_qualid`
+     `extra scopes`
+       defines extra argument scopes, to be used in case of coercion to ``Funclass``
+       (see the :ref:`implicitcoercions` chapter) or with a computed type.
+     `simpl nomatch`
+       prevents performing a simplification step for :n:`@smart_qualid`
+       that would expose a match construct in the head position.  See :tacn:`simpl`.
+     `simpl never`
+       prevents performing a simplification step for :n:`@smart_qualid`.  See :tacn:`simpl`.
+
+       .. todo: simpl nomatch/simpl never apply only to tactics simpl and cbn, or to others
+          as well (based on how simpl nomatch is described in the tactics chapter)?
+     `clear bidirectionality hint`
+       removes the bidirectionality hint, the `&`
+
+     :n:`@more_implicits_block`
+       use to specify alternative implicit argument declarations
+       for names of functions, inductive types, constructors and lemmas that can only be
+       applied to a fixed number of arguments (excluding, for instance,
+       functions whose type is polymorphic).
+       For parsing, the longest list of implicit arguments matching the function application
+       is used to select which implicit arguments are inserted.
+       For printing, the alternative with the most implicit arguments is used; the
+       implict arguments will be omitted if :flag:`Printing Implicit` is not set.
+       See the example :ref:`here<example_more_implicits>`.
+
+       .. todo the above feature seems a bit unnatural and doesn't play well with partial
+          application.  See https://github.com/coq/coq/pull/11718#discussion_r408841762
+
+       .. todo: above, are "..., inductive types, constructors and lemmas" correct even though
+          they're not functions?  I wonder why the earlier text referred to constants.
```

Comment:
Once again, just replace all of this by "function" as I think that this will make things simpler and clearer.